test/testscript.go

```diff
@@ -0,0 +1,155 @@
+package test
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+	"text/template"
+
+	"github.com/rogpeppe/go-internal/testscript"
+)
+
+type ProjectTest struct {
+	t          *testing.T
+	params     testscript.Params
+	setupFuncs []func(*testscript.Env) error
+	projectDir string
+}
+
+func NewProjectTest(t *testing.T, projectDir string) *ProjectTest {
+	params := testscript.Params{
+		Files:               []string{},
+		Setup:               nil,
+		RequireExplicitExec: true,
+		RequireUniqueNames:  true,
+	}
+	return &ProjectTest{
+		t:          t,
+		params:     params,
+		projectDir: projectDir,
+	}
+}
+
+func (e *ProjectTest) CopyFile(srcRelPath string) *ProjectTest {
+	setupFunc := func(env *testscript.Env) error {
+		src := filepath.Join(e.projectDir, srcRelPath)
+		dst := filepath.Join(env.WorkDir, srcRelPath)
+		dstDir := filepath.Dir(dst)
+		if err := os.MkdirAll(dstDir, 0755); err != nil {
+			msg := "failed to create destination directory %s: %w"
+			return fmt.Errorf(msg, dstDir, err)
+		}
+
+		srcContent, err := os.ReadFile(src)
+		if err != nil {
+			return fmt.Errorf("failed to read source file %s: %w", src, err)
+		}
+
+		if err := os.WriteFile(dst, srcContent, 0644); err != nil {
```

Comment:
I picked a permission mode of 644 here, for the files getting sent to the testscript workdir. We could also surface the permissions higher up, or pick a more restrictive set. LMK if you have any thoughts.

Reply:
Probably fine since the owner has rw and it's all temporary anyway